torch/fx/experimental/proxy_tensor.py

```diff
@@ -585,7 +586,7 @@ def unwrap_proxy(self, e):
             return get_proxy_slot(e, self, e, lambda e: e.proxy)
         elif isinstance(e, (torch.SymInt, torch.SymFloat, torch.SymBool)):
             return get_proxy_slot(e, self, e, lambda e: e())
-        elif isinstance(e, torch.ScriptObject):
+        elif isinstance(e, (torch.ScriptObject, AbstractScriptObject)):
```

Comment:
There are two purposes for AbstractScriptObject: 
1. Having all fake object wrapped by a AbstractScriptObject allows us to unify their types so that we could have a unified logic to handle all of them when necessary. For example, this `isinstance` call is to match the type of the object and retrieve its corresponding proxy with a different get_proxy_slot lambda. If we allow the class to vary for different scriptObject, we would have to isinstance check against the whole custom class registry or modify get_proxy_slot to return None or something when it cannot find a match. I feel they're not as clean as AbstractScriptObject.

2. Suppose we don't have this AbstractScriptObject wrapper. To turn method calls in to higher order op call, we need to override the original method with a new call_torchbind wrapper. However, in fake tensor mode, we actually want to unwrap the call_torchbind and call the origianal method. This requires us to store the original method somewhere and switch them back when some condition meets. This introduces a dependency with the method and the mode we're in. With this wrapper, we can easily call the original method by calling `wrappe_obj.foo` directly. This is a local change that's safer and more robust I think.

Reply:
If we wanted to, we could work around those two thing :P (For (1) we can require users subclass AbstractScriptObject, For (2) we can do some python metaclass thing to make it work)

But the approach in this PR seems fine.